sizeOptimizer.cpp

```diff
@@ -161,7 +165,85 @@ size_t SizeOptimizer::getMinSize(GeneralTransform &tr, int maxPercDecrease, bool
   return std::max(0.f, afterPercInc);
 }
 
-std::vector<GeneralTransform> *SizeOptimizer::optimizeXYZ(GeneralTransform &tr,
+std::vector<GeneralTransform> *SizeOptimizer::optimizeXYZ_3D(GeneralTransform &tr,
```

Comment:
It seems to me that the only difference to 1D/2D case is filtering vs sorting.
Is it true?
If so, pass filtering/sorting method to general 
template<typename G>
optimizeXYZ(..., G &generator)